src/ngx_http_lua_sleep.c

```diff
@@ -72,10 +72,25 @@ ngx_http_lua_ngx_sleep(lua_State *L)
     coctx->sleep.data = coctx;
     coctx->sleep.log = r->connection->log;
 
+#ifdef HAVE_POSTED_DELAYED_EVENTS_PATCH
+    if (delay == 0) {
+        dd("posting 0 sec sleep event to head of delayed queue");
+
+        coctx->sleep.delayed = 1;
+        ngx_post_event(&coctx->sleep, &ngx_posted_delayed_events);
+
+    } else {
+        dd("adding timer with delay %lu ms, r:%.*s", (unsigned long) delay,
+           (int) r->uri.len, r->uri.data);
+
+        ngx_add_timer(&coctx->sleep, (ngx_msec_t) delay);
```

Comment:
We'd better do not duplicate this code path for `delay > 0`. It is duplicated below in the `#else` branch.

I think we still need to handle the case `delay == 0` when the patch is absent. In this case, we should emit an error log message so that the user can be notified about the lack of nginx core patch instead of silently ignoring it.